### PR TITLE
MCR-1597 refactored MCRGoogleSitemap

### PR DESCRIPTION
**Pull request by KelKx**
use Solr instead of MetadataStore.use property MCR.GoogleSitemap.SolrQuery to define Solr query.
default query delivers only published and stateless MODSObjects

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ before_script:
   - |
    echo "BUILD START: $(date -u +'%Y-%m-%dT%H:%M:%SZ')"
 script:
-  - mkdir ${HOME}/tmp
+  - mkdir ${HOME}/tmp
```
Comment (yagee-de) [inline .travis.yml]: please remove this change. Has nothing to do with the ticket

File: mycore-base/src/main/resources/config/mycore.properties [hunk 2]
```diff
@@ -28,6 +28,12 @@
   MCR.baseurl=http://localhost:8291
 
 ##############################################################################
+# Google Sitemap                                                             #
+##############################################################################
+# Sets the filter for the Solr query
+  MCR.GoogleSitemap.SolrQuery=(objectType:mods AND -state:*) OR (objectType:mods AND state:published)
```
Comment (yagee-de) [inline mycore-base/src/main/resources/config/mycore.properties]: Why is that in mycore-base? mycore-base does not know anything about SOLR or site maps or even MODS.

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on ab1debffd2f5937b95420a0587d67a9923a30430! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/1635-pull_request-264

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on ab1debffd2f5937b95420a0587d67a9923a30430! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/1635-pull_request-264

File: mycore-base/src/main/resources/deprecated.properties [hunk 3]
```diff
@@ -56,6 +56,7 @@ MCR.classifications_classification_cache_size=MCR.Persistence.Classification.Sto
 MCR.classifications_category_cache_size=MCR.Persistence.Classification.Store.CacheSize.Categ
 
 MCR.googlesitemap.types=MCR.GoogleSitemap.Types
+MCR.GoogleSitemap.Types=test
 MCR.googlesitemap.freq=MCR.GoogleSitemap.Freq
```
Comment (kkrebs) [inline mycore-base/src/main/resources/deprecated.properties]: you have to write the new property name behind the old one
MCR.GoogleSitemap.Types=MCR.GoogleSitemap.SolrQuery

File: mycore-indexing/src/main/resources/components/indexing/config/mycore.properties [hunk 4]
```diff
@@ -3,8 +3,8 @@
 # The configuration for component indexing
 ##############################################################################
 
-# MyCoRe types for indexing objects, document is default
-  MCR.GoogleSitemap.Types=test
+# Sets the filter for the Solr query
```
Comment (kkrebs) [inline mycore-indexing/src/main/resources/components/indexing/config/mycore.properties]: , list all mycore objects is default

**Review by kkrebs — APPROVED** [review]

**Pull request merged**